src/vm/comutilnative.cpp

```diff
@@ -2797,18 +3002,24 @@ FCIMPL1(INT32, ValueTypeHelper::GetHashCode, Object* objUNSAFE)
     // we munge the class index with two big prime numbers
     hashCode = typeID * 711650207 + 2506965631U;
 
-    if (CanUseFastGetHashCodeHelper(pMT))
+    // Have we already checked the ability to use fast path?
+    if (pMT->HasCheckedCanCompareBitsOrUseFastGetHashCode())
     {
-        hashCode ^= FastGetValueTypeHashCodeHelper(pMT, obj->UnBox());
-    }
-    else
-    {
-        HELPER_METHOD_FRAME_BEGIN_RET_1(obj);        
-        hashCode ^= RegularGetValueTypeHashCode(pMT, obj->UnBox());
-        HELPER_METHOD_FRAME_END();
+        if (pMT->CanCompareBitsOrUseFastGetHashCode())
+        {
+            hashCode ^= FastGetValueTypeHashCodeHelper(pMT, obj->UnBox());
+        }
+        else
+        {
+            HELPER_METHOD_FRAME_BEGIN_RET_1(obj);
+            hashCode ^= RegularGetValueTypeHashCode(pMT, obj->UnBox());
+            HELPER_METHOD_FRAME_END();
+        }
+
+        return hashCode;
     }
-    
-    return hashCode;
+
+    FC_INNER_RETURN(INT32, GetHashCodeHelper(hashCode, pMT, obj));
```

Comment:
I think you can just use regular `HELPER_METHOD_FRAME_BEGIN` without `FC_INNER_RETURN`. `HELPER_METHOD_FRAME` is used in other places in this method, and so using it in one more place on a slow path won't slow it down.